IPython/core/pylabtools.py

```diff
@@ -261,6 +262,8 @@ def find_gui_and_backend(gui=None, gui_select=None):
     if gui and gui != 'auto':
         # select backend based on requested gui
         backend = backends[gui]
+        if gui == 'agg':
+            gui = None
```

Comment:
This is needed because of an error in ipykernel, where it doesn't recognize `'agg'` as a GUI and it errors if it doesn't know what GUI to use. None indicates "no GUI", which is correct in principle, but there may be a better place to put that information.

We could modify something either here or in ipykernel to make it easier to specify "this backend needs no GUI integration", which would be true for nbagg, inline, agg.